File: trunk/example/csharp/manfilter.cs
/**
 *  \file       example/csharp/manfilter.cs
 *  \brief      \$Id: manfilter.cs 507 2009-11-28 15:18:46Z dev1usr $
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev: 507 $
 *  \author     EMail: aotto1968 at users.sourceforge.net
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */
using System;
using csmsgque;
namespace example {
  sealed class manfilter : MqS, IFilterFTR {
    void IFilterFTR.Call () {
      SendSTART();
      while (ReadItemExists()) {
	SendC("<" + ReadC() + ">");
      }
      SendFTR(10);
    }
    static void Main(string[] argv) {
      manfilter srv = new manfilter();
      try {
	srv.ConfigSetName("ManFilter");
	srv.LinkCreate(argv);
	srv.ProcessEvent(MqS.WAIT.FOREVER);
      } catch (Exception ex) {
        srv.ErrorSet (ex);
      }
      srv.Exit();
    }
  }
}

File: trunk/example/vb/manfilter.vb
REM
REM  \file       example/vb/manfilter.vb
REM  \brief      \$Id: manfilter.vb 507 2009-11-28 15:18:46Z dev1usr $
REM  
REM  (C) 2009 - NHI - #1 - Project - Group
REM  
REM  \version    \$Rev: 507 $
REM  \author     EMail: aotto1968 at users.sourceforge.net
REM  \attention  this software has GPL permissions to copy
REM              please contact AUTHORS for additional information
REM

Imports System
Imports csmsgque
Imports System.Collections.Generic

Public Module example
  Private Class manfilter
    Inherits MqS
    Implements IFilterFTR

    Public Sub FilterFTR() Implements csmsgque.IFilterFTR.Call
      SendSTART()
      While ReadItemExists()
        SendC("<" + ReadC() + ">")
      End While
      SendFTR(10)
    End Sub
  End Class

  Sub Main(ByVal args() As String)
    Dim srv As New manfilter()
    Try
      srv.ConfigSetName("ManFilter")
      srv.LinkCreate(args)
      srv.ProcessEvent(MqS.WAIT.FOREVER)
    Catch ex As Exception
      srv.ErrorSet(ex)
    Finally
      srv.Exit()
    End Try
  End Sub
End Module
